test/test.js

```diff
@@ -1,243 +1,306 @@
+    RECOMMENDATION_SERVICE_ADDR,
+    grpc.credentials.createInsecure()
+  );
+  recommend = promisify(recommendationClient.listRecommendations).bind(
+    recommendationClient
+  );
+
+  const shippingClient = new hipstershop.ShippingService(
+    SHIPPING_SERVICE_ADDR,
+    grpc.credentials.createInsecure()
+  );
+  shippingQuote = promisify(shippingClient.getQuote).bind(shippingClient);
+  shippingOrder = promisify(shippingClient.shipOrder).bind(shippingClient);
+});
 
 // --------------- Ad Service ---------------
 
-test('ad: get', async t => {
-  const req = data.ad
-  const res = await adsGet(req)
+test("ad: get", async (t) => {
+  const req = data.ad;
+  const res = await adsGet(req);
 
-  t.is(res.ads.length, 2)
-  t.truthy(res.ads[0].redirectUrl)
-  t.truthy(res.ads[1].redirectUrl)
-  t.truthy(res.ads[0].text)
-  t.truthy(res.ads[1].text)
-})
+  t.is(res.ads.length, 2);
+  t.truthy(res.ads[0].redirectUrl);
+  t.truthy(res.ads[1].redirectUrl);
+  t.truthy(res.ads[0].text);
+  t.truthy(res.ads[1].text);
+});
 
 // --------------- Cart Service ---------------
 
-test('cart: all', async t => {
-  const req = data.cart
-  const userIdReq = { userId: req.userId }
+test("cart: all", async (t) => {
+  const req = data.cart;
+  const userIdReq = { userId: req.userId };
 
   // Empty Cart
-  let res = await cartEmpty(userIdReq)
-  t.truthy(isEmpty(res))
+  let res = await cartEmpty(userIdReq);
+  t.truthy(isEmpty(res));
 
   // Add to Cart
-  res = await cartAdd(req)
-  t.truthy(isEmpty(res))
+  res = await cartAdd(req);
+  t.truthy(isEmpty(res));
 
   // Check Cart Content
-  res = await cartGet(userIdReq)
-  t.is(res.items.length, 1)
-  t.is(res.items[0].productId, req.item.productId)
-  t.is(res.items[0].quantity, req.item.quantity)
+  res = await cartGet(userIdReq);
+  t.is(res.items.length, 1);
+  t.is(res.items[0].productId, req.item.productId);
+  t.is(res.items[0].quantity, req.item.quantity);
 
   // Empty Cart
-  res = await cartEmpty(userIdReq)
-  t.truthy(isEmpty(res))
+  res = await cartEmpty(userIdReq);
+  t.truthy(isEmpty(res));
 
   // Check Cart Content
-  res = await cartGet(userIdReq)
-  t.truthy(isEmpty(res))
-})
+  res = await cartGet(userIdReq);
+  t.truthy(isEmpty(res));
+});
 
 // --------------- Currency Service ---------------
 
-test('currency: supported', async t => {
-  const res = await currencySupported({})
-  t.is(res.currencyCodes.length, 33)
-})
+test("currency: supported", async (t) => {
+  const res = await currencySupported({});
+  t.is(res.currencyCodes.length, 33);
+});
 
-test('currency: convert', async t => {
-  const req = data.currency
+test("currency: convert", async (t) => {
+  const req = data.currency;
 
-  const res = await currencyConvert(req)
-  t.is(res.currencyCode, "CAD")
-  t.is(res.units, 442)
-  t.is(res.nanos, 599380805)
-})
+  const res = await currencyConvert(req);
+  t.is(res.currencyCode, "CAD");
+  t.is(res.units, 442);
+  t.is(res.nanos, 599380804);
```

Comment:
this is interesting. this might be system dependent because I added this and it was passing...
This might need to be changed to allow some wiggle room on either side
```suggestion
  t.true(res.nanos >= 599380800);
  t.true(res.nanos < 599380809);
```
something like this for example..